cmd/sops/main.go

```diff
@@ -593,7 +593,6 @@ func main() {
 				for _, path := range c.Args() {
 					err := updatekeys.UpdateKeys(updatekeys.Opts{
 						InputPath:   path,
-						GroupQuorum: c.Int("shamir-secret-sharing-quorum"),
```

Comment:
I think this should have used the option `shamir-secret-sharing-threshold`. My guess is that this was simply forgotten in 63708c6485bed2be41063a4a101d275e9c8e62fb, resp. that the option was renamed in parallel to a PR that added/modified this, and thus the old name got re-introduced here. (Turns out the PR was created after merging the rename: #255. If you look at the commits though, you can see that the first commit which already made use of the old name was created before the rename PR got merged.)

Please re-add `GroupQuorum` with the correct option value.

Reply:
I can do that, but just to double-check: My understanding is that `updatekeys` should use the `.sops.yaml` config file as the source for key updates. Since the config file has the option to set the Shamir threshold through the `shamir_threshold` key, shouldn't the user be encouraged to use that? 
 
I think allowing both the `shamir_threshold` key in the config file and the `shamir-secret-sharing-threshold` command line option would require prompting the user to resolve the conflict if both are set. Currently, that is not implemented. In fact, the shamir_threshold in .sops.yaml was not respected at all, which seem to run counter to the intention of the `updatekeys` command in the first place. 

Furthermore, neither `shamir-secret-sharing-threshold` nor `shamir-secret-sharing-quorum` are on the list of allowed options for the command. This led me to believe that the option is not meant to be in use. 

IMHO the implementation and use of `updatekeys` is simpler and more consistent when relying on the configuration file only for the Shamir threshold, but I can try to reintroduce (and rename) the `shamir-secret-sharing-threshold` option, but adding:
* `shamir-secret-sharing-threshold` to the list of allowed options
* notification to the user about the conflict, if both the option and the configuration key are set

Your input on this will be greatly appreciated.

Reply:
> Just to double-check: My understanding is that `updatekeys` should use the `.sops.yaml` config file as the source for key updates. Since the config file has the option to set the Shamir threshold through the `shamir_threshold` key, shouldn't the user be encouraged to use that?

Well, I kind of agree, but the updatekeys feature was originally implemented to explicitly allow overriding the Shamir threshold on the command line. (Unfortunately, or fortunately, depending on your POV, that implementation didn't work, due to the option rename.)

One reason is probably that the command is called `updatekeys`, and not `update-according-to-sops.yaml` or something like that. It's also not adjusting other options from `.sops.yaml` (about which parts to encrypt etc.). So there is a reason to allow overriding the Shamir threshold from the command line IMO. Even if it's not too strong.

@getsops/maintainers any opinions on this one? I'm somewhat torn between "let's fix this as it was originally intended to be" and "let's remove it since it never worked anyway, and .sops.yaml should be the single source of truth for updatekeys, also for other things than keys".

> I think allowing both the `shamir_threshold` key in the config file and the `shamir-secret-sharing-threshold` command line option would require prompting the user to resolve the conflict if both are set and differ.

I don't think so - the command line option is an override, if it's supplied the value from `.sops.yaml` is ignored. The same happens in quite a few other cases for other subcommands. (The Shamir threshold is somewhat special since you cannot explicitly *disable* it fom the command line, you can only enable it.)

Reply:
I think your reasoning is sound, @felixfontein. Reintroducing the option on the command line is a much smaller change. Using the `shamir_threshold` in the configuration "feels close" to updating the key groups, but I agree it is just on the spectrum of other things that could be considered by the `updatekeys` command, which would have been increasingly misnamed. 

I think I'll revert to fix the option for now, and that a new `update-according-to-sops-yaml` command of sorts would be a better solution to what I was trying to achieve.

Reply:
Sounds good! (And we definitely have to find a better name than `update-according-to-sops-yaml` if we're going to add that one :D )